May dev update (#105)
irods/irods.org: content/posts/irods-development-update-may-2025.md

Title: iRODS Development Update: May 2025
Date: 2025-05-19 10:00
Author: Kory Draughn
Slug: irods-development-update-may-2025
Status: published

Hello reader,

> iRODS 5 is almost ready for prime time. As of this development update, all issues have been closed and we have entered the testing phase. Let's talk about what's changed since last month's update.

Reviewer: ```suggestion
iRODS 5 is almost ready for prime time. As of this development update, all issues have been closed and we have entered the confirmation testing phase. Let's talk about what's changed since last month's update.
```

Author: Done.

> For the server, work to make the server permanently enforce strict ACLs is complete. We've also improved logging for the database plugin and GenQuery1 by replacing all rodsLog API calls with the new logging API. This change to logging gives administrators visibility into how the database plugin and GenQuery1 work. To add to that, GenQuery1 will have its very own log category. Work to remove the server's dependency on the service account irods_environment.json file has been merged. More APIs and options have been removed too.

Reviewer: ```suggestion
For the server, work to make the server permanently enforce strict ACLs is complete. We've also improved logging for the database plugin and GenQuery1 by replacing all rodsLog API calls with the new logging API. This change to logging gives administrators visibility into how the database plugin and GenQuery1 work. To add to that, GenQuery1 will have its very own log category. Work to remove the server's dependency on the service account `irods_environment.json` file has been merged. More deprecated APIs and options have been removed as well.
```

Author: Done.

For packaging, we've updated policy requirements for CMake and replaced some externals packages with system-provided packages.

To wrap up, it was reported that the Logical Quotas rule engine plugin did not cover the touch API PEPs as intended. That has been fixed. Not only that, but a new release of Metalnx is on the way. This next release focuses on improving stability.

> Thanks for reading and don't forget to register for the [User Group Meeting](https://irods.org/ugm2025/).

Reviewer: ```suggestion
Thanks for reading and don't forget to register for the [User Group Meeting](https://irods.org/ugm2025/).  The abstracts for accepted talks are now posted and available.
```

Author: Done.
